backend/src/controller/chatController.ts

```diff
@@ -102,45 +149,49 @@ async function handleHigherLevelChat(
 		transformedMessage
 			? combineTransformedMessage(transformedMessage)
 			: message,
-		transformedMessage ? true : false,
-		sentEmails,
 		currentLevel
 	);
 
 	// run defence detection and chatGPT concurrently
-	const [defenceReport, openAiReply] = await Promise.all([
-		triggeredDefencesPromise,
+	const [inputDefenceReport, openAiReply] = await Promise.all([
+		triggeredInputDefencesPromise,
 		openAiReplyPromise,
 	]);
 
-	// if input message is blocked, restore the original chat history and add user message (not as completion)
-	if (defenceReport.isBlocked) {
-		updatedChatHistory = [
-			...updatedChatHistory,
-			{
-				completion: null,
-				chatMessageType: CHAT_MESSAGE_TYPE.USER,
-				infoMessage: message,
-			},
-		];
-		updatedChatResponse = {
-			...updatedChatResponse,
-			defenceReport,
-		};
-	} else {
-		updatedChatHistory = openAiReply.chatHistory;
-		updatedChatResponse = {
-			...updatedChatResponse,
-			reply: openAiReply.chatResponse.completion?.content?.toString() ?? '',
-			wonLevel: openAiReply.chatResponse.wonLevel,
-			openAIErrorMessage: openAiReply.chatResponse.openAIErrorMessage,
-			defenceReport,
-		};
+	const botReply = openAiReply.chatResponse.completion?.content?.toString();
+	const outputDefenceReport = botReply
+		? detectTriggeredOutputDefences(botReply, defences)
+		: null;
+
+	const defenceReports = [chatResponse.defenceReport, inputDefenceReport];
```

Comment:
this logic deals with combining the defence reports, which was removed in Heather's branch [here](https://github.com/ScottLogic/prompt-injection/pull/743#discussion_r1452096240)